### PR TITLE
fix: allow not onboarded users to interact with `insights` page

### LINKED ISSUE #1645: Suggestion: "insights" tab should indicate onboarding needs to be done
### Describe the bug

When I go to `beta.insights.opensauced.pizza` I notice that the "Insights" tab is missing:

Screenshot 2023-08-31 at 2 07 40 PM

Same thing on Chrome.

I even tried checking out `v1.62.0` and `v1.61.0` and running it locally to see if it's a recent release that broke. But those are both fine locally.

But it looks totally fine in prod 🤔 

 
Screenshot 2023-08-31 at 2 22 36 PM

I'm not seeing anything obvious in the console logs that would point me to what's going on.

### Steps to reproduce

? ? ?

### Browsers

Chrome, Firefox

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #1728: Feature: Add Insights Page button shows while you are adding an insights page.
### Type of feature

🍕 Feature

### Current behavior

Screen Shot 2023-09-20 at 2 07 11 PM

### Suggested solution

Remove the button from the subheader section when already on the page. 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR introduces a new feature that enforces the completion of the onboarding process on the insights page. 

### What's New
- insights link now shows in nav links for users that have not completed onboarding
- clicking on `insights` for non-onboarded users forces them to complete their onboarding process
- refactored the step for implementation to leverage `queryParams` for more flexible control and optimization

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #1645 
fixes https://github.com/open-sauced/app/issues/1724
fixes #1728 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/62995161/6b539e82-7e66-426f-848c-5a868e9d86a2

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
I don't this is the correct solution. Why redirect them?

I think that is unnecessary friction and the "complete onboarding" button is enough.

**Comment by OgDev-01** [comment]
> I don't this is the correct solution. Why redirect them?
> 
> I think that is unnecessary friction and the "complete onboarding" button is enough.

What expected behavior do you propose for this? Users who have not completed onboarding don't have interests and can't create insights pages

**Comment by bdougie** [comment]
Two options

1. Decouple interests from insight page creation.
2. Let the user know we need interests selected before proceeding. 

We are redirecting without an explanation. We could add a toast notification as the easy 3rd option.

But this will be annoying for the user as is.

**Comment by jpmcb** [comment]
I agree with @bdougie - I think getting redirected on clicking the "Insights" tab is pretty confusing and not the expected behavior.

In my opinion, the quickfix and best solution for now is 1) that Brian mentioned which would totally decouple creating / viewing insights pages from the onboarding experience.

We could enhance this behavior in the future by prompting users during insight page creation to complete the onboarding process. But we shouldn't completely hijack their flow to redirect them to the onboarding page.

**Comment by OgDev-01** [comment]
Insights creation flow for users who have not completed onboarding
Screenshot 2023-09-20 at 17 52 28

Screenshot 2023-09-20 at 18 39 25

**Comment by OgDev-01** [comment]
Update: @bdougie and @jpmcb, users who have not completed onboarding are now allow to create insights pages

**Comment by OgDev-01** [comment]
Also fixed https://github.com/open-sauced/app/issues/1724 along the line

File: components/organisms/InsightPage/InsightPage.tsx [hunk 1]
```diff
@@ -81,7 +81,9 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
   }, [repoListData, router.query.selectedRepos, pageHref]);
 
   const { data, addMember, deleteMember, updateMember } = useInsightMembers(insight?.id || 0);
-  const { data: recommendedRepos } = useFetchInsightRecommendedRepositories();
+  const { data: recommendedRepos, isLoading } = useFetchInsightRecommendedRepositories();
+
+  console.log(recommendedRepos);
```
Comment (jpmcb) [inline components/organisms/InsightPage/InsightPage.tsx:86]: nit: Leftover `console.log`?

File: components/organisms/InsightPage/InsightPage.tsx [hunk 2]
```diff
@@ -277,6 +284,7 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
 
       if (response.ok) {
         const addedRepo = (await response.json()) as DbRepo;
+        console.log(addedRepo);
```
Comment (jpmcb) [inline components/organisms/InsightPage/InsightPage.tsx:287]: nit: Leftover `console.log`?

File: components/organisms/InsightPage/InsightPage.tsx [hunk 3]
```diff
@@ -425,6 +433,8 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
     updateSuggestionsDebounced();
   }, [repoSearchTerm]);
 
+  console.log(repos);
```
Comment (jpmcb) [inline components/organisms/InsightPage/InsightPage.tsx:436]: nit: Leftover `console.log`?

**Review by jpmcb — APPROVED** [review]
Looks good to me! I think the user flow makes sense and we can always enhance it latter to poke people to complete the onboarding

File: components/organisms/InsightPage/InsightPage.tsx (hunk 1 above)
Comment (bdougie) [reply components/organisms/InsightPage/InsightPage.tsx:86]:
```suggestion
```

File: components/organisms/InsightPage/InsightPage.tsx (hunk 2 above)
Comment (bdougie) [reply components/organisms/InsightPage/InsightPage.tsx:287]:
```suggestion
```

File: components/organisms/InsightPage/InsightPage.tsx (hunk 3 above)
Comment (bdougie) [reply components/organisms/InsightPage/InsightPage.tsx:436]:
```suggestion
```

**Comment by brandonroberts** [comment]
@OgDev-01 build errors need resolving for QA testing

**Review by bdougie — APPROVED** [review]

**Pull request merged**